### PR TITLE
Add Kotlin Wrapper for Spark Routes

### PR DESCRIPTION
**Pull request by lallemupp**
Add a Kotlin wrapper for Spark routes
to enable Kotlin syntatic sugar

### CODE REVIEW

**Comment by lallemupp** [comment]
Should the Routes class be renamed to Http?

**Comment by perwendel** [comment]
yes, let's change it to Http and we'll also add static API access

File: src/main/kotlin/spark/Routes.kt [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+class RouteHandler(val request: Request, val response: Response) : RequestHandler()
+class FilterHandler(val request: Request, val response: Response) : RequestHandler()
+
+/**
+ * The route class that takes a Spark service and wraps the route methods to enable fancy syntax
+ * with access to request and response parameters in the route code.
+ *
+ * @para service The [Spark] service that will be wrapped.
+ */
+class Routes(val service: Service) {
```
Comment (perwendel) [inline src/main/kotlin/spark/Routes.kt]: ren to `Http`

File: src/main/kotlin/spark/Routes.kt [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+open class RequestHandler {
+    val start: Long = System.currentTimeMillis()
+
+    fun requestTime(): Long {
+        return System.currentTimeMillis() - start
+    }
+}
+
+class RouteHandler(val request: Request, val response: Response) : RequestHandler()
```
Comment (perwendel) [inline src/main/kotlin/spark/Routes.kt]: add methods for direct implicit access of request and response methods that are meaningful, e.g.:
```java
    params(":name")
    queryParams("q")
    body()
    ...
    etc
```

**Pull request merged**